pages/lists/[listId]/activity.tsx

```diff
@@ -1,9 +1,103 @@
+import { createPagesServerClient } from "@supabase/auth-helpers-nextjs";
+import { GetServerSidePropsContext } from "next";
+import Error from "components/atoms/Error/Error";
+import { fetchApiData, validateListPath } from "helpers/fetchApiData";
 import ListPageLayout from "layouts/lists";
+import MostActiveContributorsCard, {
+  ContributorStat,
+} from "components/molecules/MostActiveContributorsCard/most-active-contributors-card";
 
-const ListsActivity = () => {
-  return <div>List Activity</div>;
+import useMostActiveContributors from "lib/hooks/api/useMostActiveContributors";
+import ClientOnly from "components/atoms/ClientOnly/client-only";
+
+interface ContributorListPageProps {
+  list?: DBList;
+  numberOfContributors: number;
+  isError: boolean;
+  activityData: {
+    contributorStats: { data: ContributorStat[]; meta: Meta };
+    topContributor: ContributorStat;
+  };
+}
+
+export const getServerSideProps = async (ctx: GetServerSidePropsContext) => {
+  const supabase = createPagesServerClient(ctx);
+
+  const {
+    data: { session },
+  } = await supabase.auth.getSession();
+  const bearerToken = session ? session.access_token : "";
+  const { listId } = ctx.params as { listId: string };
+  const range = 30;
+  const [
+    { data, error: contributorListError },
+    { data: list, error },
+    { data: mostActiveData, error: mostActiveError },
+  ] = await Promise.all([
+    fetchApiData<PagedData<DBListContributor>>({
+      path: `lists/${listId}/contributors?limit=1`,
+      bearerToken,
+      pathValidator: validateListPath,
+    }),
+    fetchApiData<DBList>({ path: `lists/${listId}`, bearerToken, pathValidator: validateListPath }),
+    fetchApiData<PagedData<ContributorStat>>({
+      path: `lists/${listId}/stats/most-active-contributors?range=${range}&orderDirection=DESC&orderBy=total_contributions&limit=20&contributorType=all`,
+      bearerToken,
+      pathValidator: validateListPath,
+    }),
+  ]);
+
+  if (error?.status === 404 || error?.status === 401) {
```

Comment:
Return the file not found page if the list doesn't exist or if the user is not authorized to view it.